src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -888,5 +888,157 @@ public C(string s)
+        {
+            throw new ArgumentNullException(nameof(array));
+        }
+
+        return array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count();
+    }
+}", ignoreTrivia: false);
+        }
+
+        [WorkItem(21501, "https://github.com/dotnet/roslyn/issues/21501")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInArrowExpression2()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public int Foo(int[] array[||]) /* Bar */ => /* Bar */
+        array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count(); /* Bar */
+}",
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public int Foo(int[] array) /* Bar */
+    {
+        if (array == null)
+        {
+            throw new ArgumentNullException(nameof(array));
+        }
+        /* Bar */
+        return array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count(); /* Bar */
+    }
+}", ignoreTrivia: false);
+        }
+
+        [WorkItem(21501, "https://github.com/dotnet/roslyn/issues/21501")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestMissingInArrowExpression1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public void Foo(string bar[||]) =>
+#if DEBUG
+        Console.WriteLine(""debug"" + bar);
+#else
+        Console.WriteLine(""release"" + bar);
+#endif
+}");
+        }
+
+        [WorkItem(21501, "https://github.com/dotnet/roslyn/issues/21501")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestMissingInArrowExpression2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public int Foo(int[] array[||]) =>
+#if DEBUG
+        array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count();
+#else
+        array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count();
+#endif
+}");
+        }
+
+        [WorkItem(21501, "https://github.com/dotnet/roslyn/issues/21501")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestInArrowExpression3()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public void Foo(int[] array[||]) =>
+        array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count();
+}",
+@"
+using System;
+using System.Linq;
+
+class C
+{
+    public void Foo(int[] array)
+    {
+        if (array == null)
+        {
+            throw new ArgumentNullException(nameof(array));
+        }
+
+        array.Where(x => x > 3)
+            .OrderBy(x => x)
+            .Count();
+    }
+}", ignoreTrivia: false);
```

Comment:
:memo: `ignoreTrivia` is now the default :smile:

Reply:
:memo: To be clear here, `ignoreTrivia` now defaults to *false*, meaning trivia will be compared during tests.

Reply:
@zaytsev-victor Actually it looks like the option was later removed altogether in #21745, so the build will fail until these uses are updated.